File: Data Structures & Algorithms/top-k-elements-in-list/submission-5.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        counter = {}

        for num in nums:
            counter[num] = counter.get(num, 0) + 1

        counter_list = []

        for num, freq in counter.items():
            counter_list.append([freq,num])

        counter_list.sort()

        answer = []

        while len(answer) < k:
            answer.append(counter_list.pop()[1])

        return answer

File: Data Structures & Algorithms/majority-element-ii/submission-5.cpp
class Solution {
public:
    vector<int> majorityElement(vector<int>& nums) {
        vector<int> res;
        unordered_map<int,int> counter;

        for (int num:nums) {
            counter[num] += 1;
        }

        for (auto [num,freq] : counter ) {
            if (freq > (nums.size() / 3)) {
                res.push_back(num);
            }
        }

        return res;
    }
};